File: examples/typescript/mastra/chatbot/agent.ts
/**
 * Minimal chatbot built with Mastra deployed on Superserve.
 */

import { Agent } from "@mastra/core/agent"
import { createInterface } from "readline"

const agent = new Agent({
  name: "assistant",
  instructions: "You are a helpful assistant.",
  model: "openai/gpt-4o",
})

const rl = createInterface({ input: process.stdin })
rl.on("line", async (line) => {
  const result = await agent.generate(line)
  console.log(result.text)
})

File: examples/python/pydantic-ai/chatbot/agent.py
"""
Minimal chatbot built with Pydantic AI deployed on Superserve.
"""

from pydantic_ai import Agent

agent = Agent("openai:gpt-4o", system_prompt="You are a helpful assistant.")

while True:
    try:
        user_input = input()
    except EOFError:
        break
    result = agent.run_sync(user_input)
    print(result.output)
